sqld/src/database/write_proxy/replication.rs

```diff
@@ -305,4 +308,29 @@ impl ReadReplicationHook {
 
         Ok(())
     }
+
+    #[allow(dead_code)]
+    fn is_valid_new_gen(&mut self, gen: u64, gen_start_frame_index: u64) -> bool {
+        if self.current_gen != gen {
+            if let Some(last_applied) = self.last_applied_index {
+                if last_applied <= gen_start_frame_index {
+                    // TODO persist to file.
+                    self.current_gen = gen;
```

Comment:
I think it would be cleaner to update current_gen outside of this function. Same way restart should also be triggered outside based on the result of this function.